### PR TITLE
Expand docs

### PR DESCRIPTION
**Pull request by jmalkin**
Significant expansion of our documentation!

I also realized our kolmogorov-smirnov test didn't handle items sketches, so that's also fixed.  I added improved docstrings to the generic base classes so that those would show up nicer in IDEs and on the website.

The left rail on the rendered site is a little messy, but it's probably tolerable for now. However imperfect, I think this is still a big improvement.

### CODE REVIEW

**Comment by jmalkin** [comment]
Part of issue #25

File: docs/source/hyper_log_log.rst [hunk 1]
```diff
@@ -7,26 +7,49 @@ If the ONLY use case for sketching is counting uniques and merging, the HLL sket
 This implementation offers three different types of HLL sketch, each with different trade-offs with accuracy, space and performance. 
 These types are specified with the target_hll_type parameter.
 
-In terms of accuracy, all three types, for the same lg_config_k, have the same error distribution as a function of n, the number of unique values fed to the sketch.
-The configuration parameter `lg_config_k` is the log-base-2 of `K`, where `K` is the number of buckets or slots for the sketch.
+In terms of accuracy, all three types, for the same lg_config_k, have the same error distribution as a function of `n`, the number of unique values fed to the sketch.
+The configuration parameter `lg_config_k` is the log-base-2 of `k`, where `k`` is the number of buckets or slots for the sketch.
```
Comment (c-dickens) [inline docs/source/hyper_log_log.rst:11]: For consistency with the files above, is it worth switching the backticks to double backticks? eg `k` to ``k``

**Review by c-dickens — APPROVED** [review]
Just check the backticks in frequent items sketch?

File: docs/source/hyper_log_log.rst (hunk 1 above)
Comment (jmalkin) [reply docs/source/hyper_log_log.rst]: This is done

**Pull request merged**